workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Jobs/CodegenJob.cs

```diff
@@ -18,20 +19,24 @@ public abstract class CodegenJob
         public List<string> OutputFiles = new List<string>();
         public readonly string OutputDirectory;
 
+        protected Logger logger { get; }
+
         protected readonly Dictionary<string, string> Content = new Dictionary<string, string>();
 
         private IFileSystem fileSystem;
         private readonly DetailsStore detailsStore;
 
-        public CodegenJob(string baseOutputDirectory, IFileSystem fileSystem, DetailsStore detailsStore)
+        public CodegenJob(string baseOutputDirectory, IFileSystem fileSystem, DetailsStore detailsStore, Logger logger)
         {
             OutputDirectory = baseOutputDirectory;
             this.fileSystem = fileSystem;
             this.detailsStore = detailsStore;
+            this.logger = logger;
         }
 
         public void Clean()
         {
+            logger.Info("Cleaning output directories");
```

Comment:
Prefer logging when we actually remove a directory (with the directory that was removed). That information is more useful that a log which says `Clean()` was called